src/Lean/Util/CollectFVars.lean

```diff
@@ -35,9 +35,24 @@ mutual
     | _                    => id
 end
 
+partial def collectBackwardDeps (lctx : LocalContext) (i : Nat) : Visitor := fun s =>
+  if h : i < s.fvarIds.size then
+    let s := if let some ldecl := lctx.find? s.fvarIds[i] then
+      visit ldecl.type s
```

Comment:
Missing `ldecl.value?`. 
It will not affect the current PR since we don't use let-decls in inductive declarations, but it may affect future uses of `collectBackwardDeps`.